pkg/metadata/provider_vmware.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"path"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	guestMetaData = "guestinfo.metadata"
+
+	guestUserData = "guestinfo.userdata"
+)
+
+// ProviderVMware is the type implementing the Provider interface for VMware
+type ProviderVMware struct {
+	cmd string
+}
+
+// NewVMware returns a new ProviderVMware
+func NewVMware() *ProviderVMware {
+	return &ProviderVMware{}
+}
+
+func (p *ProviderVMware) String() string {
+	return "VMWARE"
+}
+
+// Probe checks if we are running on VMware
+func (p *ProviderVMware) Probe() bool {
+	c, err := exec.LookPath("vmware-rpctool")
```

Comment:
Will this really be available if built with linuxkit? Wouldn't this tool only be available in the open-vm-tools image?

Reply:
It will be available if open-vm-tools is intalled, which is sort of a requirement if you want to make RPC calls through VMCI to get guestinfo data.... I dont understand what is being requested.

This mirrors the functionality of vmware's cloud-init https://github.com/vmware/cloud-init-vmware-guestinfo/blob/master/DataSourceVMwareGuestInfo.py which functions the same way.

Reply:
Since edvinerikson has brought it up I had checked if it would be possible not to use the command-line tool and use a library instead. It is possible to use this [library](https://github.com/vmware/vmw-guestinfo). But this pull request has been idle for so long I am not sure maintainers have any interest in accepting it. I do not want to work on it if it is not going to be ignored.

Reply:
I made a small PR to your source branch, as an example, with the said library 8 days ago: https://github.com/brlbil/linuxkit/pull/1
This would give you the possibility of removing the vmware-rpctool